### PR TITLE
Remove percent_indicator lib

### PR DESCRIPTION
**Pull request by BenjaminBelanger**
### ⁉️ Related Issue
Closes: #1075 

## 📖 Description
Replaces `percent_indicator` library that is no longer maintained with native Flutter class `CircularProgressIndicator`

percent_indicator was used for:
- Student grade circular display
- Program completion circular indicator in profile view

### 🧪 How Has This Been Tested?
- Tested on Android
- Tested that the new widgets look the same as previously
- Tested that grade-based color schemes remain the same
- Tested that animations look the same
- Tested with horizontal and vertical phone orientation

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] If needed, I added analytics.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`. 

### 🖼️ Screenshots (if useful):

https://github.com/user-attachments/assets/51187e89-bd5c-479d-a21a-253ceb829959

https://github.com/user-attachments/assets/11dd6bb2-5347-440a-a416-1f08478aeab0

### CODE REVIEW

File: lib/features/student/grades/widgets/grade_circular_progress.dart [hunk 1]
```diff
@@ -59,30 +58,53 @@ class _GradeCircularProgressState extends State<GradeCircularProgress>
...
+        duration: Duration(milliseconds: duration),
+        curve: Curves.easeOut,
+        builder: (_, value, __) => CircularProgressIndicator(
+          value: value,
+          strokeWidth: 8.0 * widget.ratio,
+          strokeCap: StrokeCap.round,
+          backgroundColor: Colors.grey[300],
+          valueColor: AlwaysStoppedAnimation<Color>(color),
+        ),
       ),
```
Comment (XavierPaquet-Rapold) [inline lib/features/student/grades/widgets/grade_circular_progress.dart:107]: Mettre ça dans un widget séparé

File: lib/features/student/profile/profile_view.dart [hunk 2]
```diff
@@ -276,46 +275,48 @@ Card getMyBalanceCard(ProfileViewModel model, BuildContext context) {
 
 Card getProgramCompletion(ProfileViewModel model, BuildContext context) {
```
Comment (XavierPaquet-Rapold) [inline lib/features/student/profile/profile_view.dart:276]: Créer un widget contenant cet méthode

**Review by XavierPaquet-Rapold — CHANGES_REQUESTED** [review]

File: lib/features/student/grades/widgets/grade_circular_progress.dart (hunk 1 above)
Comment (BenjaminBelanger) [reply lib/features/student/grades/widgets/grade_circular_progress.dart]: J'ai créé le nouveau fichier **grade_circular_indicator.dart** bd33e5b28f532e39e8ef6a438de3c5bee6e649f4

File: lib/features/student/profile/profile_view.dart (hunk 2 above)
Comment (BenjaminBelanger) [reply lib/features/student/profile/profile_view.dart]: Je l'ai mise dans un nouveau fichier **program_completion.dart** 5327ed9dd880538ec1bf770756757d9d34ba72cd

**Review by XavierPaquet-Rapold — APPROVED** [review]
LGTM 🔥

**Review by EffBureau — APPROVED** [review]
LGTM 🤓

**Pull request merged**